sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSViewModel.kt

```diff
@@ -44,12 +44,9 @@ sealed interface MerchantDetailsState {
 class POSViewModel : ViewModel() {
     private val _uiState = MutableStateFlow(POSUIState())
     val uiState: StateFlow<POSUIState> = _uiState.asStateFlow()
-    private val api: PosApiService by lazy {
-        // lazy because we may not have the merchantId
-        // of the forageConfig when a ViewModel instance
-        // is created but we certainly will by the time
-        // we ever use any methods of the api
-        PosApiService.from(uiState.value.posForageConfig)
+
+    private fun getForageRetrofitApi(): PosApiService {
+        return PosApiService.from(uiState.value.posForageConfig)
     }
```

Comment:
```suggestion
    private val api
        get() = PosApiService.from(uiState.value.posForageConfig)
```
Nice fix! I'm embarrassed that I introduced this bug without evening realizing it. Thanks for identifying the problem and fixing 💛 .

Throwing out this suggestion if you want because it maaaaaay be a more Kotlin-y way to do this. I've been getting the impression that kotlin seems to opt for .accessors over `getters` / `setters`. But I haven't taken the time to read through conventions so I could be full of sh*t. I'm going to approve the PR since this fix works!

Reply:
Nice! Updated to this approach in [d643a4f](https://github.com/teamforage/forage-android-sdk/pull/201/commits/d643a4f23b0a59a82c375196a4b5452f412f81f9)